packages/grid/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -389,6 +389,7 @@ export const GridRootStyles = styled('div', {
     },
     [`.${c.menuOpen}`]: {
       visibility: 'visible',
+      width: 'auto',
```

Comment:
This line was removed, do you remember the reason?
It caused this issue:

https://github.com/mui/mui-x/assets/13808724/4f9a2db7-a2c6-4300-89ae-4a1a6a75a80b

Reply:
Not sure. The menu icons were messing with the flex+sticky layout because in their hidden state their were sticking outside at the end of the row, so they were adding width and widening the scroll container more than it needed to be. I modified them until it didn't happen, don't remember about this specific line.

Reply:
I've added it back, let me know if it's causing other issues.